traits/ctraits.c

```diff
@@ -4841,9 +4845,13 @@ get_trait_post_setattr(trait_object *trait, void *closure)
 static int
 set_trait_post_setattr(trait_object *trait, PyObject *value, void *closure)
 {
-    if (!PyCallable_Check(value)) {
+    if(value == Py_None){
+        value = NULL;
+    }
+
+    if (value != NULL && !PyCallable_Check(value)) {
         PyErr_SetString(
-            PyExc_ValueError, "The assigned value must be callable.");
+            PyExc_ValueError, "The assigned value must be callable or None.");
```

Comment:
So I think we need to do more here: if we're setting `None` as a value for `py_post_setattr`, then having `trait->post_setattr` be `post_setattr_trait_python` is wrong. Instead, we want `trait->post_setattr` to be `NULL` as well.

The invariant should be that if `trait->post_setattr` is `post_setattr_trait_python`, then `py_post_setattr` is non-NULL.